### PR TITLE
[collectd 6] Allow arbitrary UTF-8 strings as label names.

### PR DESCRIPTION
**Pull request by octo**
Source: https://opentelemetry.io/docs/specs/semconv/general/attribute-naming/

ChangeLog: Daemon: All valid UTF-8 strings are now accepted as resource attribute name and metric label name.

### CODE REVIEW

**Review by eero-t — CHANGES_REQUESTED** [review]
I'd like test(s) checking also handling of extra (closing) curly brace(s).  Otherwise looks OK on quick look.

**Comment by octo** [comment]
> I'd like test(s) checking also handling of extra (closing) curly brace(s). Otherwise looks OK on quick look.

Done.

**Comment by octo** [comment]
Rebased to resolve a merge conflict.

File: src/daemon/metric.c [hunk 1]
```diff
@@ -29,20 +29,23 @@
...
-  "0123456789_.-"
+  "0123456789_.-:"
 
 /* Metric names must match the regex `[a-zA-Z_:][a-zA-Z0-9_:]*` */
 // instrument-name = ALPHA 0*254 ("_" / "." / "-" / "/" / ALPHA / DIGIT)
-#define VALID_NAME_CHARS VALID_LABEL_CHARS "/"
+#define VALID_NAME_CHARS                                                       \
+  "abcdefghijklmnopqrstuvwxyz"                                                 \
+  "ABCDEFGHIJKLMNOPQRSTUVWXYZ"                                                 \
+  "0123456789_.-/"
```
Comment (eero-t) [inline src/daemon/metric.c:48]: After `git grep abcdefghijklmnopqrstuvwxyz`, I wonder whether it would make sense to add `ALNUM_CHARS` define that would be used by both of these:
```
#define UNQUOTED_LABEL_CHARS  ALNUM_CHARS "_.-:"
#define VALID_NAME_CHARS      ALNUM_CHARS "_.-/"
```

And also in:
* `write_prometheus.c`: `VALID_LABEL_CHARS` / `VALID_NAME_CHARS`
* `format_stackdriver.c`: `metric_type()`::`valid_chars`

But that could be another PR.

**Review by eero-t — APPROVED** [review]
Approved (tests being added and them passing makes this easier!)
